File: qualifier-week-5/graded-assignments/GrPA2 - Graded.c
#include <stdio.h>

// Write code below
int GCD(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() {
    int a,b;
    scanf("%d",&a);
    scanf("%d",&b);
    printf("%d",GCD(a,b));
    return 0;
}

/* Shouldn't it Work ? But not Working

#include <stdio.h>

int GCD(int a, int b) {
  if (b == 0) {
    return a;
  } else {
    return GCD(b, a % b);
  }
}

int main() {
  int n1, n2;

  scanf("%d", n1);
  scanf("%d", n2);

  if (n1 <= 0 || n2 <= 0) {
    printf("Error!: Failure");
    return 1;
  }

  int gcd = GCD(n1, n2);
  printf("%d\n", gcd);

  return 0;
}

File: qualifier-week-5/graded-assignments/GrPA2 - Graded.rs
use std::io;

fn main() {
    let mut input1 = String::new();
    io::stdin().read_line(&mut input1).expect("Error: Failure");
    let a: i32 = input1.trim().parse().expect("Error Input!: Failure");

    let mut input2 = String::new();
    io::stdin().read_line(&mut input2).expect("Error: Failure");
    let a: i32 = input2.trim().parse().expect("Error Input!: Failure");

    println!(gcd(input1, input2));
}

fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b != 0 {
        let temp = a;
        b = a % b;
        a = temp;
    }
    return a;
}